pkg/auth/providers/azure/azure_provider_test.go

```diff
@@ -0,0 +1,192 @@
+package azure
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/rancher/norman/api/writer"
+	"github.com/rancher/norman/types"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	managementschema "github.com/rancher/rancher/pkg/schemas/management.cattle.io/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestConfigureTest inspects the Redirect URL during Azure AD setup.
+func TestConfigureTest(t *testing.T) {
```

Comment:
There doesn't appear to be a test for our new functionality to extract annotations when they are provided directly. Would you mind adding a test which takes the annotations from the top level?

Reply:
As a side note: there's other test cases that would be good to have here as well like:
- What happens if there's a key for annotations but it isn't of type map[string]interface{}?
- What happen if we have our target key but it's value isn't a string type?

That being said, these aren't the main purpose of this PR so they are more of nice-to-haves.

Reply:
Yes, there is a problem with test setup for this case. With stubbing the Norman schemas for the test. I did spend time on trying to chase that down, but could not figure out why having no `metadata` field causes Norman's schema validation to discard annotations as a superfluous field. This does not happen when the code runs outside of tests.

Reply:
Met offline to clear up the issue. Thank you, Michael!